Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -873,6 +941,73 @@ namespace xr
             pose.Position.Y = rawPose[5];
             pose.Position.Z = rawPose[6];
         }
+
+        /**
+         * Checks whether this plane has been subsumed (i.e. no longer needed), and adds it to the vector if so.
+         **/
+        void CheckForSubsumedPlanes(std::vector<Frame::Plane::Identifier>& subsumedPlanes)
+        {
+            auto planeMapIterator = planeMap.begin();
+            while (planeMapIterator != planeMap.end())
+            {
+                auto [arPlane, planeID] = *planeMapIterator;
+
+                // Check if the plane has been subsumed, and if we should stop tracking it.
+                ArPlane* subsumingPlane = nullptr;
+                ArPlane_acquireSubsumedBy(session, arPlane, &subsumingPlane);
+
+                // Plane has been subsumed, stop tracking it explicitly.
+                if (subsumingPlane != nullptr)
+                {
+                    subsumedPlanes.push_back(planeID);
+                    Planes[planeID].Polygon.clear();
+                    Planes[planeID].PolygonSize = 0;
+                    planeMapIterator = planeMap.erase(planeMapIterator);                    
```

Comment:
nit: If this works, it works, but it seems a little weird. I think it's more typical to collect a list of keys to erase than to erase from a collection while you're actively iterating over it.